src/smart_pentest_framework/core/decision_engine.py

```
@@ -0,0 +1,16 @@
+class DecisionEngine:
+    def __init__(self, logger):
+        self.logger = logger
+
+    def suggest_next_steps(self, results):
+        self.logger.info("Analyzing results to suggest next steps...")
+        suggestions = []
+        if "recon" in results:
+            for port_info in results["recon"]:
+                port = int(port_info["port"])
+                service = port_info["service"]
```

Ruff (0.12.2)

Local variable `service` is assigned to but never used

Remove assignment to unused variable `service`

(F841)